### ISSUE TITLE
eth_signTypedData not sending result to browser

### ISSUE THREAD

**Issue by 1blockologist** [MetaMask/metamask-extension#4993]
www.pareto.network not accessible with Metamask with today's version 

The console log mentions that "This experimental version of eth_signTypedData will be deprecated", is there a new function we should be using, or is it the same method name with different logic inside? Since this message is still there and not actually deprecated yet, then this is a bug that needs a hot fix

anything I can help with?

**Comment by 1blockologist** [comment]
I feel like a unit test would have noticed this

**Comment by evansjohnson** [comment]
I am also having this issue. For me, just before the MM UI updates to reflect a spinner that never resolves, the message `"cannot read property EIP712Domain of undefined"` shows on the UI.

I found the same message in a failed test of `eth-sig-util`:
https://circleci.com/gh/MetaMask/eth-sig-util/48#config/containers/0
https://github.com/MetaMask/eth-sig-util/pull/20
which was resolved by using the renamed method, `signTypedDataLegacy`

Perhaps the extension is calling signTypedData?  Still looking into this will update if I find anything else of interest.
